pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -555,19 +555,92 @@ func inconsistentStatus(set *apps.StatefulSet, status *apps.StatefulSetStatus) b
 		status.UpdateRevision != set.Status.UpdateRevision
 }
 
-// completeRollingUpdate completes a rolling update when all of set's replica Pods have been updated
-// to the updateRevision. status's currentRevision is set to updateRevision and its' updateRevision
-// is set to the empty string. status's currentReplicas is set to updateReplicas and its updateReplicas
+// completeUpdate completes an update when all of set's replica Pods have been updated
+// to the updateRevision. status's currentRevision is set to updateRevision.
+// status's currentReplicas is set to updateReplicas and its updateReplicas
 // are set to 0.
-func completeRollingUpdate(set *apps.StatefulSet, status *apps.StatefulSetStatus) {
-	if set.Spec.UpdateStrategy.Type == apps.RollingUpdateStatefulSetStrategyType &&
-		status.UpdatedReplicas == status.Replicas &&
+func completeUpdate(set *apps.StatefulSet, status *apps.StatefulSetStatus) {
+	if status.UpdatedReplicas == status.Replicas &&
```

Comment:
@dbason this seems weird. I think this should be `AvailableReplicas` now and it must have been missed when graduating `StatefulSetMinReadySeconds`.